### PR TITLE
Added Support for Automatic Retry 

### PR DESCRIPTION
**Pull request by dalwadi2**
Create RetryHelper Functionality for poor network APIs which gives different error code due to network fluctuation.

### CODE REVIEW

**Comment by dodgex** [comment]
You are using `package harsh.dalwadi.retrofitretryhelper;` in most of your files. ;)

**Comment by PromanSEW** [comment]
Please add copyright headers in all files and fix package names

**Comment by dalwadi2** [comment]
@dodgex I will correct that one.
@PromanSEW I don't know about Travis very well

**Comment by dalwadi2** [comment]
@PromanSEW can you guide me with that?

**Comment by PromanSEW** [comment]
@dalwadi2 you can remove imports because it is the same package
See https://travis-ci.org/square/retrofit/builds/233555375#L272-L334 and look at existing code at repository for undestanding code style

File: samples/src/main/java/com/example/retrofit/SimpleRetryService.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2012 Square, Inc.
```
Comment (PromanSEW) [inline samples/src/main/java/com/example/retrofit/SimpleRetryService.java]: It is 2017 now

**Comment by dalwadi2** [comment]
@PromanSEW solved.
any other suggestion?
or in which file I have to change now?

**Comment by dalwadi2** [comment]
Still, Travis checks failed.
@PromanSEW can you suggest me what do I have to change?

File: retrofit/src/main/java/retrofit2/CustomCallback.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package retrofit2;
+
+import Call;
```
Comment (PromanSEW) [inline retrofit/src/main/java/retrofit2/CustomCallback.java]: These "imports" are unnecessary and break build. Same for other files. Please remove them

**Comment by PromanSEW** [comment]
Also you forgot copyright headers

**Comment by dalwadi2** [comment]
@PromanSEW 
what about copyright headers? can you explain to me what to do?

File: retrofit/src/main/java/retrofit2/CustomCallback.java (hunk 2 above)
Comment (dalwadi2) [inline retrofit/src/main/java/retrofit2/CustomCallback.java]: removed

**Comment by PromanSEW** [comment]
See https://github.com/square/retrofit/blob/master/retrofit/src/main/java/retrofit2/Call.java#L1-L15 for example

**Comment by dalwadi2** [comment]
@PromanSEW thanks for your Support

**Comment by dalwadi2** [comment]
@JakeWharton can you please tell me the reason for closing this? or any Suggestion that how I make better this functionality?
Thanks.

**Comment by JakeWharton** [comment]
I echo Jesse's comments. See also https://github.com/square/retrofit/issues/2180#issuecomment-279124009.

**Comment by dalwadi2** [comment]
Okay, thanks.

**Comment by WingSafari5** [comment]
Is this (or similar) behaviour available in OkHTTP?
Also I think Square and their people are doing a grea job at providing high-quality librarys for the android ecosystem.
Android just wouldn't be the same w/o you!

**Pull request closed without merging**